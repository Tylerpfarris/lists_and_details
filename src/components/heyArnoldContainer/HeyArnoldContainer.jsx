import React, { Component } from 'react'
import { fetchCharacters } from '../../services/heyArnoldApi'
import CharacterList from '../characters/CharacterList';


export default class HeyArnoldContainer extends Component {
    state = {
        loading: true,
        characters: [],
    }

    async componentDidMount() {
        try {
            const characters = await fetchCharacters();
        this.setState({
            characters,
            loading: false,
        });
        } catch (error) {
            throw Error(error)
        }
    
    }

    render() {
        const { loading, characters } = this.state;
        return (loading) ? <h1>Loading...</h1>:
            <CharacterList characters={characters} name='' id='' />
    }
}
