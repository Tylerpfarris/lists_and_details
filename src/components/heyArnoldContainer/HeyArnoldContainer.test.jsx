import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import HeyArnoldContainer from './HeyArnoldContainer';


describe('HeyArnoldContainer', () => {
    it('renders a list of characters to the page', async () => {
        render(
            <MemoryRouter>
               <HeyArnoldContainer /> 
            </MemoryRouter>
            );

        screen.getByText('Loading...')
        const ul = await screen.findByRole('list', { name: 'characters' });
        expect(ul).toMatchSnapshot()
    })
});