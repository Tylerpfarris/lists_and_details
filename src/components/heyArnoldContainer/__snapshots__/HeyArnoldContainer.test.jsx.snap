// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeyArnoldContainer renders a list of characters to the page 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f5c"
    >
      <label
        style="font-size: 40px;"
      >
        Nancy
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f59"
    >
      <label
        style="font-size: 40px;"
      >
        Grandpa's grandfather
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f52"
    >
      <label
        style="font-size: 40px;"
      >
        Grandpa Phil
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f58"
    >
      <label
        style="font-size: 40px;"
      >
        Grandpa's father
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f70"
    >
      <label
        style="font-size: 40px;"
      >
        Phoebe Heyerdahl
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f7e"
    >
      <label
        style="font-size: 40px;"
      >
        Eugene Horowitz
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f82"
    >
      <label
        style="font-size: 40px;"
      >
        Ray
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f8b"
    >
      <label
        style="font-size: 40px;"
      >
        Shelley
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f76"
    >
      <label
        style="font-size: 40px;"
      >
        Lila Sawyer
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f7c"
    >
      <label
        style="font-size: 40px;"
      >
        Uncle Stinky
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f55"
    >
      <label
        style="font-size: 40px;"
      >
        Arnie
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f66"
    >
      <label
        style="font-size: 40px;"
      >
        Olga Pataki
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f6b"
    >
      <label
        style="font-size: 40px;"
      >
        Timberly Johanssen
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f6d"
    >
      <label
        style="font-size: 40px;"
      >
        Jerry Berman
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f73"
    >
      <label
        style="font-size: 40px;"
      >
        Rhonda Wellington Lloyd
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f78"
    >
      <label
        style="font-size: 40px;"
      >
        Stinky Peterson
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f71"
    >
      <label
        style="font-size: 40px;"
      >
        Kyo Heyerdahl
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f8e"
    >
      <label
        style="font-size: 40px;"
      >
        Iggy
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f77"
    >
      <label
        style="font-size: 40px;"
      >
        Mr. Sawyer
      </label>
    </a>
  </li>
  <li>
    <a
      href="/details/5da237699734fdcb7bef8f84"
    >
      <label
        style="font-size: 40px;"
      >
        Thaddeus "Curly" Gammelthorpe
      </label>
    </a>
  </li>
</ul>
`;
